test/e2e/storage/csi_snapshot.go

```diff
@@ -0,0 +1,254 @@
+			storageClass := newStorageClass(storageClassTest, ns.GetName(), "")
+			By("creating a StorageClass " + storageClass.Name)
+			storageClass, err := cs.StorageV1().StorageClasses().Create(storageClass)
+			Expect(err).NotTo(HaveOccurred())
+			defer func() {
+				framework.Logf("deleting storage storageClass %s", storageClass.Name)
+				framework.ExpectNoError(cs.StorageV1().StorageClasses().Delete(storageClass.Name, nil))
+			}()
+
+			By("creating a claim")
+			claim := newClaim(storageClassTest, ns.GetName(), "", nil)
+			claim.Spec.StorageClassName = &storageClass.ObjectMeta.Name
+			claim, err = cs.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(claim)
+			Expect(err).NotTo(HaveOccurred())
+			defer func() {
+				framework.Logf("deleting claim %q/%q", claim.Namespace, claim.Name)
+				// typically this claim has already been deleted
+				err = cs.CoreV1().PersistentVolumeClaims(claim.Namespace).Delete(claim.Name, nil)
+				if err != nil && !apierrs.IsNotFound(err) {
+					framework.Failf("Error deleting claim %q. Error: %v", claim.Name, err)
+				}
+			}()
+			err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, cs, claim.Namespace, claim.Name, framework.Poll, framework.ClaimProvisionTimeout)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking the claim")
+			// Get new copy of the claim
+			claim, err = cs.CoreV1().PersistentVolumeClaims(claim.Namespace).Get(claim.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Get the bound PV
+			pv, err := cs.CoreV1().PersistentVolumes().Get(claim.Spec.VolumeName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			snapshotClassTest := driver.createSnapshotClassTest(node)
+			snapshotClass := newSnapshotClass(snapshotClassTest, ns.GetName(), "scc")
+			By("creating a SnapshotClass")
+			snapshotClass, err = dynamicClient.Resource(snapshotClassGVR).Create(snapshotClass, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			defer func() {
+				framework.Logf("deleting SnapshotClass %s", snapshotClass.GetName())
+				framework.ExpectNoError(dynamicClient.Resource(snapshotClassGVR).Delete(snapshotClass.GetName(), nil))
+			}()
+
+			By("creating a snapshot")
+			snapshot := newSnapshot(claim, ns.GetName(), snapshotClass.GetName())
+			snapshot, err = dynamicClient.Resource(snapshotGVR).Namespace(ns.GetName()).Create(snapshot, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			defer func() {
+				framework.Logf("deleting snapshot %q/%q", snapshot.GetNamespace(), snapshot.GetName())
+				// typically this snapshot has already been deleted
+				err = dynamicClient.Resource(snapshotGVR).Namespace(ns.GetName()).Delete(snapshot.GetName(), nil)
+				if err != nil && !apierrs.IsNotFound(err) {
+					framework.Failf("Error deleting snapshot %q. Error: %v", claim.Name, err)
+				}
+			}()
+			WaitForSnapshotReady(dynamicClient, snapshot.GetNamespace(), snapshot.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
```

Comment:
check the error returned from WaitForSnapshotReady?

Reply:
Done